[#268] add message error handling (#315)
streamnative/pulsar-io-cloud-storage: README.md

      batchTimeMs: 1000
    ```

### Data format types

> Cloud Storage Sink Connector provides multiple options of output format, which are JSON, Avro, Bytes, or Parquet. The default format is JSON.

Reviewer: ```suggestion
Cloud Storage Sink Connector provides multiple output format options, including JSON, Avro, Bytes, or Parquet. The default format is JSON.
```

With current implementation, there are some limitations for different formats:

> - JSON: only support structured Pulsar schema types, including `JSON`, `PROTOBUF`, `AVRO`, and `PROTOBUF_NATIVE`, will try to convert the data to JSON from `String` and `Bytes` message.

Reviewer: ```suggestion
- JSON: only support structured Pulsar schema types, including `JSON`, `PROTOBUF`, `AVRO`, and `PROTOBUF_NATIVE`. The connector will try to convert the data with a `String` or `Bytes` format to JSON-format data.
```

- Avro / Parquet: only support structured Pulsar schema types, including `JSON`, `PROTOBUF`, `AVRO`, and `PROTOBUF_NATIVE`.
- Bytes: support all Pulsar schema types.

> When the connector receiving a message with a schema type that is not supported by the connector, the connector will `fail` the message by default, if you want to skip the non-supported messages, you can set `skipFailedMessages` to `true`.

Reviewer: ```suggestion
By default, when the connector receives a message with a non-supported schema type, the connector will `fail` the message. If you want to skip the non-supported messages, you can set `skipFailedMessages` to `true`.
```

> ### Dead letter topic

Reviewer: ```suggestion
### Dead-letter topics
```

> To use dead-letter-topic, you need to set `skipFailedMessages` to `false`, and set `--max-redeliver-count` and `--dead-letter-topic` when submit the connector with `pulsar-admin`. For more info about dead letter topic, please refer to the [Pulsar documentation](https://pulsar.apache.org/docs/en/concepts-messaging/#dead-letter-topic).

Reviewer: ```suggestion
To use a dead-letter topic, you need to set `skipFailedMessages` to `false`, and set `--max-redeliver-count` and `--dead-letter-topic` when submit the connector with the `pulsar-admin` CLI tool. For more info about dead-letter topics, see the [Pulsar documentation](https://pulsar.apache.org/docs/en/concepts-messaging/#dead-letter-topic).
```

> When dead-letter-topic is enabled, the connector will send the message to the dead-letter-topic when the message is failed to be sent to the Cloud Storage.

Reviewer: ```suggestion
If a message fails to be sent to the Cloud Storage and there is a dead-letter topic, the connector will send the message to the dead-letter topic.
```
